### ISSUE TITLE
Cairo Context error in executable? 

### ISSUE THREAD

**Issue by Finalfantasykid** [pyinstaller/pyinstaller#5147]
I am getting the following error when running the built executable:

`TypeError: Couldn't find foreign struct converter for 'cairo.Context'`

I don't get the error when running it within the virtual environment, only after the I run with the compiled binary.  The error occurs in both Windows and Linux.  And for some context, this is a GTK application which has a Surface which has some basic shapes drawn to it, but this doesn't happen with the error.

The relevant imports I have are:
```python
from gi import require_version
require_version('Gtk', '3.0')
from gi.repository import Gtk, Gdk, GdkPixbuf, GLib
```

And I have the cairo context being passed from the 'draw' signal of a GtkDrawingArea:
```xml
<object class="GtkDrawingArea" id="overlay">
    <property name="visible">True</property>
    <property name="can_focus">False</property>
    <signal name="draw" handler="drawOverlay" swapped="no"/>
</object>
```

The signature of the drawOverlay function is:
```python
def drawOverlay(self, widget, cr):
```

I tried a couple things to make sure that everything was imported, like adding `import cairo` and `from cairo import Context` but neither helped.  Is there a hidden import or binary I need to add in the pyinstaller command?

EDIT: Also I should mention I have tried with version 3.6 and 4.0 of pyinstaller, both have produce the error.

**Comment by rokm** [comment]
Reproduced on Fedora 32 (python 3.8.5, gtk 3.24.22, `develop` pyinstaller) and following program:

```python
#!/usr/bin/env python3

from gi import require_version
require_version('Gtk', '3.0')
from gi.repository import Gtk, Gdk, GdkPixbuf, GLib

class App (Gtk.Window):
    def __init__ (self):
        super().__init__()

        self.connect("destroy", Gtk.main_quit)

        self.draw_area = Gtk.DrawingArea()
        self.draw_area.connect("draw", self.on_draw)
        self.add(self.draw_area)

        self.show_all()

    def on_draw (self, widget, cairo_context):
        print(f"Cairo context: {cairo_context}")

app = App()
Gtk.main()
```

**Comment by rokm** [comment]
As far as I can tell, the issue is that in order to properly mix Cairo objects between `gi` wrapper (i.e., `from gi.repository import cairo`) and `pycairo` wrapper (i.e., `import cairo`), the conversion function in `gi` must be available and loaded. (Apparently the `cairo_context` comes from `pycairo` not `gi`, hence it is "foreign object").

When running from source, this is taken care of automatically. In built version, this does not happen (perhaps we are missing some internal import that should trigger it).

As a work-around, you can perform the registration manually:
```python
import gi
import cairo
gi.require_foreign('cairo')
```

Adding this block to the above example fixes the error for me.

**Comment by rokm** [comment]
Hmm, actually, with the above example, just adding 
```python
import cairo
```

seems to suffice. Or alternatively, running `pyinstaller --hidden-import cairo ...`.

**Comment by Finalfantasykid** [comment]
I tried your example and got the same results.  But when I try the same with my application, the problem persists.   I'm using a glade xml file for the UI, so I'm not sure if that has anything to do with it.  I tried adding the signal in the code like you did, but that didn't fix it.

https://github.com/Finalfantasykid/AstraStack

That's the code that I'm trying to build.  UI.py is where the relevant code should be add.
